src/Command/CheckCommand.php

```diff
@@ -186,4 +191,48 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         return $exitCode;
     }
+
+    private function checkUpdates(InputInterface $input, OutputInterface $output): bool
+    {
+        if (!extension_loaded('Phar') || !($localPhar = \Phar::running(false))) {
+            $output->writeln('<comment>Drupal Check was not installed as a Phar archive and cannot self-update</comment>.', OutputInterface::VERBOSITY_DEBUG);
+            return false;
+        }
+        if (!is_writable($localPhar)) {
+            $output->writeln("<comment>Cannot update as the Phar file is not writable: $localPhar</comment>");
+            return false;
+        }
+
+        $currentVersion = $this->getApplication()->getVersion();
+        if ($currentVersion === Application::FALLBACK_VERSION || strpos($currentVersion, 'dev-') === 0) {
+            $output->writeln('<comment>Cannot update with a development installation.</comment>', OutputInterface::VERBOSITY_DEBUG);
+            return false;
+        }
+
+        $updater = new Updater(null, false);
+        $updater->setStrategy(Updater::STRATEGY_GITHUB);
+        $updater->getStrategy()->setPackageName('mglaman/drupal-check');
+        $updater->getStrategy()->setPharName('drupal-check.phar');
+        $updater->getStrategy()->setCurrentLocalVersion($currentVersion);
+
+        if (!$updater->hasUpdate()) {
+            return true;
+        }
+
+        $newVersion = $updater->getNewVersion();
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(sprintf('<question>Update to version %s? [Y/n]</question>', $newVersion), true);
```

Comment:
What happens if the no interaction flag is passed? Should this default to false and just display output?

Reply:
That sounds reasonable.